imap_processing/codice/codice_l1a.py

```diff
@@ -509,9 +552,11 @@ def process_codice_l1a(file_path: Path, data_version: str) -> xr.Dataset:
 
```

Comment:
I just noticed that you are only processing first packet from `packet_dataset`. Is that right?

Reply:
I am quite confused here. Since `packet_file_to_datasets()` function (which is used within my `decom_packets()` function above) packages packets into distinct datasets, one dataset per APID, wouldn't I always expect just one packet within a particular `datasets[apid]`?

Reply:
In general, I am still confused about what to expect in a typical CoDICE L0 file when we are in-flight.

My understanding is that I will receive a `imap_codice_l0_raw_YYYYMMDD_vXXX.pkts` file that will contain multiple APIDs (say, for example, `1136` (hskp) + `1155` (sw-priority) + 1160` (nsw-priority)), which will then be rearranged into:

```python
{1136 : <xr.Dataset for 1136>, 1155: <xr.Dataset for 1155>, 1160: <xr.Dataset for 1160>}
```

Where `<xr.Dataset for 1155>` for example, contains one packet with APID `1155` which has one `epoch` associated with it.

It confuses me how I could get multiple `epochs` for a single APID, as @greglucas eludes to in one of his comments above.

Am I misinterpreting anything here?

Reply:
You got it right up until this sentence `Where <xr.Dataset for 1155> for example, contains one packet with APID 1155 which has one epoch associated with it.`

`packet_file_to_datasets()` will unpack all packets you get in `imap_codice_l0_raw_YYYYMMDD_vXXX.pkts`. Let's say that your daily file has 100 packets for this particular 1155 apid, it will decom all those 100 packets per XTCE definition and store it as `xr.Dataset` where there is one variable in `xr.Dataset` for each field the XTCE definition. Therefore, all of your variable in the `xr.Dataset` has 100 element array. Therefore, your `epochs` has 100 epoch as well. I hope that helps. I am happy to jump on call if it helps.

Reply:
Ah ok, thanks for clarifying! So I was correct in thinking one packet = one epoch, but I didn't realize that the `xr.Dataset` could have many packets. I hadn't encountered that before since the test data I am using is just a single file and a single packet.

My code will require some changes to support multiple packets, but I will take care of that in a future PR that addresses #769